### ISSUE TITLE
Collection append in PHP 7.1 & CakePHP 3.5.1 leads to a collection of ArrayIterators

### ISSUE THREAD

**Issue by rsadza** [cakephp/cakephp#11226]
This is a (multiple allowed):

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 3.5.1
* Platform and Target: CLI / PhpUnit
```
PHP 7.1.9-1+ubuntu16.04.1+deb.sury.org+1 (cli) (built: Sep  2 2017 05:56:43) ( NTS )
Copyright (c) 1997-2017 The PHP Group
Zend Engine v3.1.0, Copyright (c) 1998-2017 Zend Technologies
    with Zend OPcache v7.1.9-1+ubuntu16.04.1+deb.sury.org+1, Copyright (c) 1999-2017, by Zend Technologies
```

### What you did
```php
$collection = new Collection([1, 2]);
$combined = $collection->append([3, 4]);
$combined2 = $combined->append([5, 6]);

debug($combined->toList());
debug($combined2->toList());
```
### What happened
`$combined->toList()` returns an array `[1, 2, 3, 4]` as expected.
```php
[
	(int) 0 => (int) 1,
	(int) 1 => (int) 2,
	(int) 2 => (int) 3,
	(int) 3 => (int) 4
]
``` 

But `$combined2->toList()` returns an array of 2 ArrayIterators and 5, 6:
```php
[
	(int) 0 => object(ArrayIterator) {
		
	},
	(int) 1 => object(ArrayIterator) {
		
	},
	(int) 2 => (int) 5,
	(int) 3 => (int) 6
]
``` 

### What you expected to happen
I expected that `$combined2->toList()` would return `[1, 2, 3, 4, 5, 6]`.

**Comment by markstory** [comment]
I wasn't able to reproduce this locally in PHP 7.1.8. I've pushed 0a9a0de562e90e24815389427b482ec323002822 to see if travis can reproduce the error you're seeing in any of the versions it has.

**Comment by markstory** [comment]
Travis had the same failure as you did. There is [a possibly relevant bug that was fixed on PHP 7.1.9](https://bugs.php.net/bug.php?id=74977) that may have caused this regression.

**Comment by ADmad** [comment]
I ran the new tests on PHP 7.1.10 and didn't have any failures either. So I think it's safe to say it was a PHP bug which has been fixed.
